include/arch/posix/posix_soc_if.h

```diff
@@ -3,8 +3,8 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-#ifndef ZEPHYR_ARCH_POSIX_INCLUDE_POSIX_SOC_IF_H_
-#define ZEPHYR_ARCH_POSIX_INCLUDE_POSIX_SOC_IF_H_
+#ifndef ZEPHYR_INCLUDE_ARCH_POSIX_POSIX_SOC_IF_H_
```

Comment:
And this header is not actually meant to be public for the app (is internal for the posix arch/soc/boards)
(z_arch_irq_lock / z_arch_irq_unlock should not be here..)

Reply:
@stephanosio If this header needs to be kept in a "public" place due to it being included in arch.h then that's fine

Reply:
I somehow managed to keep posix_soc_if.h private in the newly pushed commit.